libpromises/rlist.c

```diff
@@ -1657,8 +1657,12 @@ bool RlistEqual(const Rlist *list1, const Rlist *list2)
             assert(rp1->val.item == NULL && rp2->val.item == NULL);
         }
     }
-
-    return true;
+    if (rp1 == NULL && rp2 == NULL) {
+        return true;
+    } else {
+        // Lengths are different
+        return false;
+    }
```

Comment:
Could be written as a single statement
```suggestion
    // return false if lengths are different
    return (rp1 == NULL && rp2 == NULL);
```